pkg/cmd/issue/view/view.go

```diff
@@ -188,140 +194,32 @@ func printRawIssuePreview(out io.Writer, issue *api.Issue) error {
 	return nil
 }
 
-func printHumanIssuePreview(opts *ViewOptions, baseRepo ghrepo.Interface, issue *api.Issue) error {
-	out := opts.IO.Out
-	cs := opts.IO.ColorScheme()
-
-	// Header (Title and State)
-	fmt.Fprintf(out, "%s %s#%d\n", cs.Bold(issue.Title), ghrepo.FullName(baseRepo), issue.Number)
-	fmt.Fprintf(out,
-		"%s • %s opened %s • %s\n",
-		issueStateTitleWithColor(cs, issue),
-		issue.Author.Login,
-		text.FuzzyAgo(opts.Now(), issue.CreatedAt),
-		text.Pluralize(issue.Comments.TotalCount, "comment"),
-	)
+func humanIssuePreview(ipf *IssuePrintFormatter, isCommentsPreview bool) error {
```

Comment:
**Food For Thought**

At first review I think this might benefit from a bit of refactoring around ["Tell Don't Ask"](https://martinfowler.com/bliki/TellDontAsk.html). Instead of giving all this information to a thing and then asking for it back, I suspect it would form better cohesiveness if we had two concrete formatters that we could inject to satisfy an interface, and then called render on those.

I think this would be a good opportunity to pair on. I may be totally wrong, and that would be fine too but it's a nice nuanced concept that will likely appear many times during our time together on the team so spending time on it together seems like it would pay dividends.

Reply:
Ah, interesting... 🤔 I'll think on that, but pairing does sound like a good idea.